src/manager.cpp

```diff
@@ -160,13 +179,113 @@ void Manager::deleteSingleFruVpd(
     logging::logMessage(std::string(i_dbusObjPath));
 }
 
+bool Manager::isValidUnexpandedLocationCode(
+    const std::string& i_unexpandedLocationCode)
+{
+    if ((i_unexpandedLocationCode.length() <
+         constants::UNEXP_LOCATION_CODE_MIN_LENGTH) ||
+        ((i_unexpandedLocationCode.compare(0, 4, "Ufcs") !=
+          constants::STR_CMP_SUCCESS) &&
+         (i_unexpandedLocationCode.compare(0, 4, "Umts") !=
+          constants::STR_CMP_SUCCESS)) ||
+        ((i_unexpandedLocationCode.length() >
+          constants::UNEXP_LOCATION_CODE_MIN_LENGTH) &&
+         (i_unexpandedLocationCode.find("-") != 4)))
+    {
+        return false;
+    }
+
+    return true;
+}
+
 std::string Manager::getExpandedLocationCode(
-    const sdbusplus::message::object_path& i_dbusObjPath)
+    const std::string& i_unexpandedLocationCode,
+    [[maybe_unused]] uint16_t i_nodeNumber)
 {
-    // Dummy code to supress unused variable warning. To be removed.
-    logging::logMessage(std::string(i_dbusObjPath));
+    if (!isValidUnexpandedLocationCode(i_unexpandedLocationCode))
+    {
+        phosphor::logging::elog<types::InvalidArgument>(
+            types::Argument::ARGUMENT_NAME("LOCATIONCODE"),
+            types::Argument::ARGUMENT_VALUE(i_unexpandedLocationCode.c_str()));
+    }
 
-    return std::string{};
+    const nlohmann::json& l_parsedJson = m_worker->getSysConfigJSONObj();
```

Comment:
l_sysCfgJsonObj
as per new coding guidelines .

Reply:
getSysConfigJSONObj()-->API returns the parsed JSON file.
so l_parsed_Json is good.